src/SignalR/clients/csharp/Client.SourceGenerator/src/HubProxyGenerator.cs

```diff
@@ -0,0 +1,256 @@
+            var proxiesBody = new StringBuilder(@"
+namespace Microsoft.AspNetCore.SignalR.Client.SourceGenerated
+{");
+
+            foreach (var (hubSymbol, memberAccess) in hubSymbols.Values)
+            {
+                var typeName = $"Generated{hubSymbol.Name}";
+                proxiesBody.AppendLine($@"
+    public sealed class {typeName} : {hubSymbol.ToString()}
+    {{
+        private readonly Microsoft.AspNetCore.SignalR.Client.IHubConnection conn;
+        public {typeName}(Microsoft.AspNetCore.SignalR.Client.IHubConnection connection)
+        {{
+            this.conn = connection;
+        }}");
+
+                var members = hubSymbol.GetMembers()
+                    .Where(member => member.Kind == SymbolKind.Method)
+                    .Select(member => (IMethodSymbol) member)
+                    .Concat(hubSymbol.AllInterfaces.SelectMany(x => x
+                        .GetMembers()
+                        .Where(member => member.Kind == SymbolKind.Method)
+                        .Select(member => (IMethodSymbol) member)));
+
+                // Generate each of the methods
+                foreach (var member in members)
+                {
+                    var signature = new StringBuilder($"public {member.ReturnType.ToString()} {member.Name}(");
+                    var first = true;
+
+                    var callArgs = new StringBuilder("");
+
+                    // Populate signature with its parameters
+                    foreach (var parameter in member.Parameters)
+                    {
+                        if (!first)
+                        {
+                            signature.Append(", ");
+                        }
+
+                        first = false;
+                        signature.Append($"{parameter.Type.ToString()} {parameter.Name}");
+
+                        callArgs.Append($", {parameter.Name}");
+                    }
+
+                    signature.Append(')');
+
+
+                    var specificCall = getSpecificCall(member);
+                    string call;
+                    if (specificCall is null)
+                    {
+                        call = "throw new System.NotSupportedException();";
+                        context.ReportDiagnostic(Diagnostic.Create(
+                            new DiagnosticDescriptor(
+                                "XX0003",
+                                "Not supported method",
+                                "'{0}.{1}' has an unsupported signature for proxy generation.",
+                                "SignalR.Client.SourceGenerator",
+                                DiagnosticSeverity.Warning,
+                                true), memberAccess.GetLocation(), hubSymbol.Name, member.Name));
+                        // Location for the diagnostic report is the GetProxy call site since there is no guarantee that
+                        //  the method definition lays within source of current project.
```

Comment:
[Per discussion on issue](https://github.com/dotnet/aspnetcore/issues/32534#issuecomment-849805352), methods with non-`Task` and non-stream return types (e.g. `void`, `int` etc..`) produce build time warnings and calls at runtime throw `NotSupportedException`. Originally suggestion was a build time error. I felt this was too strong. However if the team feels an error is more appropriate, I'm happy to increase the severity.

Reply:
> Like "method {x} does not return a Task which is not supported for proxy generation."

Yes. Maybe ".. return a Task, ValueTask nor IAsyncEnumerable which is no .." since either of those are valid.

> For the other case, would it only be for new method types that we add to SignalR and just need to update the generator?

I assume you mean in what other cases would this warning be triggered? Yes any new method type as well as potentially some bugs in the generator logic.